src/dodal/common/data.py

```diff
@@ -0,0 +1,145 @@
+                    return DataCollectionIdentifier.parse_obj(json)
+                else:
+                    raise Exception(response.status)
+
+    async def get_current_collection(self) -> DataCollectionIdentifier:
+        async with ClientSession() as session:
+            async with session.get(f"{self._url}/numtracker") as response:
+                if response.status == 200:
+                    json = await response.json()
+                    return DataCollectionIdentifier.parse_obj(json)
+                else:
+                    raise Exception(response.status)
+
+
+class LocalVisitServiceClient(VisitServiceClientBase):
+    # VisitService API
+    """Local or dummy impl of VisitService client to co-ordinate unique filenames."""
+
+    _count: int
+
+    def __init__(self) -> None:
+        self._count = 0
+
+    async def create_new_collection(self) -> DataCollectionIdentifier:
+        self._count += 1
+        return DataCollectionIdentifier(collectionNumber=self._count)
+
+    async def get_current_collection(self) -> DataCollectionIdentifier:
+        return DataCollectionIdentifier(collectionNumber=self._count)
+
+
+class StaticVisitDirectoryProvider(DirectoryProvider):
+    """
+    Gets information from a remote service to determine how their files should be named.
+    Supports a single Visit which should be passed as a Path relative to the root of the Detector IOC mounting.
+    i.e. for a Kubernetes IOC that mounts `/dls/ixx/data` as `/data`: root="/data/YYYY/visit/"
+    but for a Detector running with a view to the whole DLS filesystem, root="/dls/ixx/data/YYYY/visit"
+    Requires that all detectors are running with a mutual view on the filesystem.
+    """
+
+    _beamline: str
+    _root: Path
+    _client: VisitServiceClientBase
+    _current_collection: DirectoryInfo | None
+    _session: ClientSession | None
+
+    def __init__(
+        self,
+        beamline: str,
+        root: Path,
+        client: Optional[VisitServiceClientBase] = None,
+    ):
+        self._beamline = beamline
+        self._client = client or VisitServiceClient(f"{beamline}-control:8088/api")
+        self._root = root
+        self._current_collection = None
+        self._session = None
+
+    async def update(self) -> None:
+        """
+        Calls the visit service to create a new data collection in the current visit.
+        """
+        # TODO: After visit service is more feature complete:
+        # TODO: Allow selecting visit as part of a request, pass down to this class
+        # TODO: Query visit service to get information about visit and data collection
+        # TODO: Use AuthN information as part of verification with visit service
+
+        try:
+            collection_id_info = await self._client.create_new_collection()
+            self._current_collection = self._generate_directory_info(collection_id_info)
+        except Exception as e:
+            self._current_collection = None
+            raise e
```

Comment:
```suggestion
        except Exception:
            self._current_collection = None
            raise
```

I think it would also be useful to debug log here, something like "got error so clearing current collection".